library/dellemc/idrac/dellemc_idrac_firmware.py

```diff
@@ -132,80 +227,266 @@
+                                                                     share_name,
+                                                                     share_user,
+                                                                     share_pwd,
+                                                                     share_mnt,
+                                                                     catalog_file_name,
+                                                                     apply_update,
+                                                                     reboot,
+                                                                     job_wait)
+
     except RuntimeError as e:
         module.fail_json(msg=str(e))
 
-    if "Status" in msg['update_status']:
-        if msg['update_status']['Status'] == "Success":
+    if "Status" in result['update_status']:
+        if result['update_status']['Status'] == "Success":
+            result['msg'] = 'Successfully created the repository update job.'
             if module.params['job_wait']:
-                msg['changed'] = True
+                remote['msg'] = 'Succesfully completed the repository update job.'
+                result['changed'] = True
         else:
-            module.fail_json(msg='Failed to update firmware.', update_status=msg['update_status'])
-    return msg
+            result['msg'] = 'Failed to update firmware.'
+            module.fail_json(**result)
+    else:
+        result['msg'] = 'Failed to update firmware.'
+        module.fail_json(**result)
+
+    return result
 
 
 def main():
+
     module = AnsibleModule(
         argument_spec={
+            # iDRAC Credentials
             "idrac_ip": {"required": True, "type": 'str'},
             "idrac_user": {"required": True, "type": 'str'},
             "idrac_pwd": {"required": True, "type": 'str', "no_log": True},
             "idrac_port": {"required": False, "default": 443, "type": 'int'},
 
+            # Network File Share
             "share_name": {"required": True, "type": 'str'},
-            "share_user": {"required": False, "type": 'str'},
-            "share_pwd": {"required": False, "type": 'str', "no_log": True},
-            "share_mnt": {"required": True, "type": 'str'},
+            "share_user": {"required": False, "type": 'str', "default": None},
+            "share_pwd": {"required": False, "type": 'str', "default": None, "no_log": True},
+            "share_mnt": {"required": False, "type": 'path', "default": None},
 
-            "catalog_file_name": {"required": False, "type": 'str', "default": "Catalog.xml"},
-            "reboot": {"required": False, "type": 'bool', "default": False},
-            "job_wait": {"required": False, "type": 'bool', "default": True},
+            # Firmware update parameters
+            "catalog_file_name": {"required": False, "default": 'Catalog.xml', "type": 'str'},
+            "apply_update": {"required": False, "default": True, "type": 'bool'},
```

Comment:
Why this option `apply_update` required to support firmware update from HTTP Source